climada/hazard/tc_surge_bathtub.py

```diff
@@ -0,0 +1,179 @@
+    """TC surge heights in m, a bathtub model with wind-surge relationship and inland decay."""
+
+    def __init__(self):
+        Hazard.__init__(self, HAZ_TYPE)
+
+
+    @staticmethod
+    def from_tc_winds(wind_haz, topo_path, inland_decay=True, add_sea_level_rise=0.0):
+        """Compute tropical cyclone surge from input winds.
+
+        Parameters
+        ----------
+        wind_haz : TropCyclone
+            Tropical cyclone wind hazard object.
+        topo_path : str
+            Path to a raster file containing gridded elevation data.
+        inland_decay : bool, optional
+            Surge decays according to distance from coast. Default: True.
+        add_sea_level_rise : float, optional
+            Sea level rise effect in meters to be added to surge height.
+        """
+        centroids = copy.deepcopy(wind_haz.centroids)
+
+        if not centroids.coord.size:
+            centroids.set_meta_to_lat_lon()
+
+        # Select wind-affected centroids which are inside INLAND_MAX_DIST_KM and |lat| < 61
+        if not centroids.dist_coast.size or np.all(centroids.dist_coast >= 0):
+            centroids.set_dist_coast(signed=True, precomputed=True)
+        coastal_msk = (((wind_haz.intensity > 0).sum(axis=0).A1 > 0)
+                       & (centroids.dist_coast < 0)
+                       & (centroids.dist_coast > -MAX_DIST_COAST * 1000)
+                       & (np.abs(centroids.lat) < 61))
+
+        # elevation at coastal centroids, exclude high centroids
+        coastal_centroids_h = coord_util.read_raster_sample(
+            topo_path, centroids.lat[coastal_msk], centroids.lon[coastal_msk])
+        elevation_msk = ((coastal_centroids_h >= 0)
+                         & (coastal_centroids_h <= MAX_ELEVATION + add_sea_level_rise))
+        coastal_msk[coastal_msk] = elevation_msk
+        coastal_centroids_h = coastal_centroids_h[elevation_msk]
+        coastal_idx = coastal_msk.nonzero()[0]
+
+        # Initialize array at coastal centroids
+        inten_surge = wind_haz.intensity.copy()
+        inten_surge[:,(~coastal_msk).nonzero()[0]] = 0
+        inten_surge.eliminate_zeros()
+
+        # Conversion of wind to surge using the linear wind-surge relationship from
+        # figure 2 of the following paper:
+        #
+        #   Xu, Liming (2010): A Simple Coastline Storm Surge Model Based on Pre-run SLOSH Outputs.
+        #   In: 29th Conference on Hurricanes and Tropical Meteorology, 10–14 May. Tucson, Arizona.
+        #   https://ams.confex.com/ams/pdfpapers/168806.pdf
+        inten_surge.data = 0.1023 * np.fmax(inten_surge.data - 26.8224, 0) + 1.8288
+
+        if inland_decay:
+            # add decay according to distance from coast
+            dist_coast_km = np.abs(centroids.dist_coast[coastal_idx]) / 1000
+            coastal_centroids_h += DECAY_RATE_M_KM * dist_coast_km
+        coastal_centroids_h -= add_sea_level_rise
+        inten_surge[:,coastal_idx] -= coastal_centroids_h[None,:]
```

Comment:
Detail: I am not 100% sure what we said on this kind of notation, but I have vague memories that Chahan just added sth to the style guide regarding avoiding this.

Reply:
This is not mere "notation". How we write this does affect the performance of the code. The in-place operations `+=`, `-=`, `*=` and so forth are very efficient and should be used as much as possible.

Reply:
Ok, good to know! @chahank , side note to your comment for the climada coding convention today, then - 
with saying
> avoid complex numpy expressions, e.g. `a = b + c * d` vs. `a = c * d; a += b` 
you mean the first should be discouraged, the second encouraged, hence? Then that was just an understanding issue from my side!

Reply:
Yes exactly. Maybe the guide should be rephrased then.